_posts/2020-05-03-dynamic-ipv6-playground.md

```diff
@@ -0,0 +1,121 @@
+categories:                    
+  - blog                       
+tags:                          
+  - ipv6                       
+---                            
+                               
+In this post we will take a tour into the dynamic IPv6 setup, examining the
+varios options and scenarios it presents.
+
+This post is a direct continuation to the
+[Static IPv6 Playground](../static-ipv6-playground) post.
+The tooling and the basic setup will remain the same.
+
+## Basic setup
+Please revisit the setup description in the
+[static IPv6 basic setup](../static-ipv6-playground/#basic-setup) section.
+
+## The Server
+An IPv6 dynamic setup consist of a Router Advertisement (RA) service and an
+optional DHCPv6 service which serves an entire network.
+
+### Router Advertisement
+In an IPv6 network, routers send RA messages that include the needed
+information for setting up a host interface
+[IPv6 configuration](https://tools.ietf.org/html/rfc4862).
+The data included in the RA message includes the network address
+(with its prefix length), routing information and potentially other extended
+network information (e.g. [DNS](https://tools.ietf.org/html/rfc8106) entries).
+
+### DHCPv6
+Although RA is sufficient for setting up a client host interface with IPv6
+configuration, DHCPv6 complements it with additional information in
+scenarios where additional options are in need.
+The latest [RFC](https://tools.ietf.org/html/rfc8415) even proposes a full
+alternative to the stateless address autoconfiguration (SLAAC), but is probably
+less common in the industry.
+
+DHCPv6 mainly complements SLAAC by allowing a specific IPv6 address to be set
+on an interface and the distribution of DNS information. It is also designed
+like its DHCPv4 origin with extendable options in mind, making it more
+accomodated for quick custom extensions.
+
+As the DNS information is already available in RA messages, DHCPv6 is mainly
+left suited for administrators that need to control the exact IPv6 full address
+(unlike SLAAC where only the network is distributed by RA and the host address
+is auto-generated).
+
+### dnsmasq
+We will use [dnsmasq](http://www.thekelleys.org.uk/dnsmasq/doc.html) to setup
+both an RA and DHCPv6 server to provide dynamic IPv6 services to the client.
+
+The server configuration is set on the *blue* namespace.
+
+- dnsmasq-config-ra-only
+```config
+leasefile-ro
+interface=veth10
+enable-ra
+dhcp-range=::,constructor:veth10,ra-only
+```
+
+- dnsmasq-config-slaac-dhcpv6
+```config
+leasefile-ro
+interface=veth10
+enable-ra
+dhcp-range=::2220,::2222,constructor:veth10,slaac
+```
+> **Note**: The DHCP server may also be configured to provide "static"
+addresses per specific identifiers (e.g. the interface mac address).
+This is useful for controlling the exact IPv6 address a requesting interface.
+With dnsmasq this translates to the following lines:
+```
+dhcp-host=12:34:56:78:90:ab,[fd00::22]
+dhcp-range=::,static
```

Comment:
"The optional keyword may be static which tells dnsmasq to enable
DHCP for the network specified, but not to dynamically allocate IP
addresses: only hosts which have static addresses given via dhcp-host or
from /etc/ethers will be served. A static-only subnet with address all
zeros may be used as a "catch-all" address to enable replies to all
Information-request packets on a subnet which is provided with stateless
DHCPv6, ie --dhcp-range=::,static"

maybe something from here worth mentioning ?
i did see you mentioned most

Reply:
I added this as part of a note about how to allocate a fixed IP address. I cannot include all the information that exists, I leave it up to the reader to explore dnsmasq specifics if s/he is interested in, for my need I think it is enough.